pkg/controller/cloneset/cloneset_controller.go

```diff
@@ -163,6 +180,12 @@ func (r *ReconcileCloneSet) doReconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, nil
 	}
 
+	// If scaling expectations have not satisfied yet, just skip this reconcile.
+	if scaleSatisfied, scaleDirtyPods := scaleExpectations.SatisfiedExpectations(request.String()); !scaleSatisfied {
```

Comment:
A dump question. If cs.Replica is changed but previous expectation has not been reached yet. Will dropping reconcile request be problematic?

Reply:
Yes. If expectations is not satisfied, it means the `create/delete/update` actions on pod or pvc have not been noticed by the informer. So this time the pods/pvcs got from the informer cache are `dirty` in some ways, we can not do reconcile depending on them.

Reply:
Thanks. I understand the expectation design pattern now.